internal/cmd/opensearch/instance/update/update.go

```diff
@@ -0,0 +1,265 @@
+				}
+				return err
+			}
+			err = req.Execute()
+			if err != nil {
+				return fmt.Errorf("update OpenSearch instance: %w", err)
+			}
+			instanceId := model.InstanceId
+
+			// Wait for async operation, if async mode not enabled
+			if !model.Async {
+				s := spinner.New(cmd)
+				s.Start("Updating instance")
+				_, err = wait.PartialUpdateInstanceWaitHandler(ctx, apiClient, model.ProjectId, instanceId).WaitWithContext(ctx)
+				if err != nil {
+					return fmt.Errorf("wait for OpenSearch instance update: %w", err)
+				}
+				s.Stop()
+			}
+
+			operationState := "Updated"
+			if model.Async {
+				operationState = "Triggered update of"
+			}
+			cmd.Printf("%s instance %s\n", operationState, instanceLabel)
+			return nil
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Bool(enableMonitoringFlag, false, "Enable monitoring")
+	cmd.Flags().String(graphiteFlag, "", "Graphite host")
+	cmd.Flags().Int64(metricsFrequencyFlag, 0, "Metrics frequency")
+	cmd.Flags().String(metricsPrefixFlag, "", "Metrics prefix")
+	cmd.Flags().Var(flags.UUIDFlag(), monitoringInstanceIdFlag, "Monitoring instance ID")
+	cmd.Flags().StringSlice(pluginFlag, []string{}, "Plugin")
+	cmd.Flags().Var(flags.CIDRSliceFlag(), sgwAclFlag, "List of IP networks in CIDR notation which are allowed to access this instance")
+	cmd.Flags().StringSlice(syslogFlag, []string{}, "Syslog")
+	cmd.Flags().Var(flags.UUIDFlag(), planIdFlag, "Plan ID")
+	cmd.Flags().String(planNameFlag, "", "Plan name")
+	cmd.Flags().String(versionFlag, "", "Instance OpenSearch version")
+
+	cmd.MarkFlagsMutuallyExclusive(planIdFlag, planNameFlag)
+	cmd.MarkFlagsMutuallyExclusive(planIdFlag, versionFlag)
+	cmd.MarkFlagsRequiredTogether(planNameFlag, versionFlag)
+}
+
+func parseInput(cmd *cobra.Command, inputArgs []string) (*inputModel, error) {
+	service := "opensearch"
+	operation := cmd.Use
+	instanceId := inputArgs[0]
+
+	globalFlags := globalflags.Parse(cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &cliErr.ProjectIdError{}
+	}
+
+	enableMonitoring := flags.FlagToBoolPointer(cmd, enableMonitoringFlag)
+	monitoringInstanceId := flags.FlagToStringPointer(cmd, monitoringInstanceIdFlag)
+	graphite := flags.FlagToStringPointer(cmd, graphiteFlag)
+	metricsFrequency := flags.FlagToInt64Pointer(cmd, metricsFrequencyFlag)
+	metricsPrefix := flags.FlagToStringPointer(cmd, metricsPrefixFlag)
+	plugin := flags.FlagToStringSlicePointer(cmd, pluginFlag)
+	sgwAcl := flags.FlagToStringSlicePointer(cmd, sgwAclFlag)
+	syslog := flags.FlagToStringSlicePointer(cmd, syslogFlag)
+	planId := flags.FlagToStringPointer(cmd, planIdFlag)
+	planName := flags.FlagToStringValue(cmd, planNameFlag)
+	version := flags.FlagToStringValue(cmd, versionFlag)
+
+	if planId != nil && (planName != "" || version != "") {
+		return nil, &cliErr.DSAInputPlanError{
+			Service:   service,
+			Operation: operation,
```

Comment:
`cmd.Use` is `"update INSTANCE_ID"`, is this intended or is just `"update"` intended?

Reply:
Yes, I think it makes sense so we can see it in the error help:
![image](https://github.com/stackitcloud/stackit-cli/assets/28832884/d98ed54b-8dd6-4e81-8fb4-32af486a180e)

Reply:
Similarly in mongodbflex:
![image](https://github.com/stackitcloud/stackit-cli/assets/28832884/5d25870a-dd53-43a2-871d-19055d7141f0)

Reply:
But we could improve even more by providing the actual instance ID

Reply:
But I would do it in a separate PR also, as it impacts more services